### PR TITLE
improve on error logic

### PR DESCRIPTION
**Pull request by slavik-lvovsky**

### CODE REVIEW

**Comment by perspectivus1** [comment]
Your solution is related to https://github.com/SAP/yeoman-ui/issues/45.

Also see this [old comment](https://github.com/SAP/yeoman-ui/blame/f9401350b019b0c9c4cd67ba3154a6d870ab3e6c/backend/src/yeomanui.ts#L158).
I think the rationale behind _not_ using the modern promise version of `Generator.run()` was that old generators would not work(??)

**Comment by slavik-lvovsky** [comment]
> Your solution is related to #45.
> 
> Also see this [old comment](https://github.com/SAP/yeoman-ui/blame/f9401350b019b0c9c4cd67ba3154a6d870ab3e6c/backend/src/yeomanui.ts#L158).
> I think the rationale behind _not_ using the modern promise version of `Generator.run()` was that old generators would not work(??)

I checked several old generator and they worked

File: backend/package.json [hunk 1]
```diff
@@ -106,7 +106,7 @@
 		"strip-ansi": "^6.0.0",
 		"titleize": "^1.0.1",
 		"ws": "^7.2.5",
-		"yeoman-environment": "2.10.0"
+		"yeoman-environment": "2.8.0"
```
Comment (perspectivus1) [inline backend/package.json:109]: Was there a regression in 2.10.0? Why downgrade?

Comment (slavik-lvovsky) [reply backend/package.json:109]: I will not downgrade

**Pull request merged**